src/config.js

```diff
@@ -0,0 +1,181 @@
+
+/**
+ * Options and settings set as inputs to this action.
+ *
+ * @see {@link ../action.yml}
+ */
+export default class Config {
+  /**
+   * Options of retries for failed requests.
+   * @readonly
+   * @enum {string} The option for retries.
+   */
+  Retries = {
+    /** No retries, just hope that things go alright.
+     * @readonly
+     */
+    ZERO: "0",
+    /**
+     * Five retries in five minutes.
+     * @readonly
+     */
+    FIVE: "5",
+    /**
+     * Ten retries in about thirty minutes.
+     * @readonly
+     */
+    TEN: "10",
+    /**
+     * A burst of retries to keep things running fast.
+     * @readonly
+     */
+    RAPID: "RAPID",
+  };
+
+  /**
+   * @typedef Inputs - Values provided to this job.
+   * @property {boolean} errors - If the job should exit after errors or succeed.
+   * @property {string?} method - The Slack API method to call.
+   * @property {string?} payload - Request contents from the provided input.
+   * @property {string?} payloadDelimiter - Seperators of nested attributes.
+   * @property {string?} payloadFilePath - Location of a JSON request payload.
+   * @property {boolean} payloadTemplated - If templated values are replaced.
+   * @property {string?} proxy - An optional proxied connection for requests.
+   * @property {Retries} retries - The retries method to use for failed requests.
+   * @property {string?} token - The authentication value used with the Slack API.
+   * @property {string?} webhook - A location for posting request payloads.
+   * @property {string?} webhookType - Posting method to use with the webhook.
+   */
+
+  /**
+   * @type {Inputs} - The actual action input values.
+   */
+  inputs;
+
+  /**
+   * @type {import("axios").AxiosStatic} - The axios client.
+   */
+  axios;
+
+  /**
+   * @type {Content} - The parsed payload data to send.
+   */
+  content;
+
+  /**
+   * Shared utilities specific to the GitHub action workflow.
+   * @type {import("@actions/core")}
+   */
+  core;
+
+  /**
+   * @type {import("@slack/web-api")} - Slack API client.
+   */
+  webapi;
+
+  /**
+   * Gather values from the job inputs and use defaults or error for the missing
+   * ones.
+   *
+   * The content of the payload is also parsed, proxies set, and a shared "core"
+   * kept for later use.
+   *
+   * @constructor
+   * @param {core} core - GitHub Actions core utilities.
+   */
+  constructor(core) {
+    this.axios = axios;
+    this.core = core;
+    this.webapi = webapi;
+    this.inputs = {
+      errors: core.getBooleanInput("errors"),
+      method: core.getInput("method"),
+      payload: core.getInput("payload"),
+      payloadDelimiter: core.getInput("payload-delimiter"),
+      payloadFilePath: core.getInput("payload-file-path"),
+      payloadTemplated: core.getBooleanInput("payload-templated") || false,
+      proxy:
+        core.getInput("proxy") ||
+        process.env.HTTPS_PROXY ||
+        process.env.https_proxy ||
+        null,
```

Comment:
It feels strange to write this, but IMO we might not need to document the environment variable usage, at least for right now?

These variables can be super useful when sharing variables throughout the whole job, but for clarity in the workflow I believe it's much better to use the direct `proxy` (or similar) input.

Super open to thoughts about this though!